Cannot compile with mingw-w64 under msys2

When trying to compile with mingw-w64 under msys2 the error: "The opendir() API is provided by this compiler, so should not be built here" is raised.

When i force off "USE_WINDOWS" the following errors are raised:
```
_deps/zipper-src/zipper/unzipper.cpp: In member function 'void zipper::Unzipper::Impl::changeFileDate(const string&, uLong, tm_unz)':
_deps/zipper-src/zipper/unzipper.cpp:196:9: error: 'HANDLE' was not declared in this scope
  196 |         HANDLE hFile;
      |         ^~~~~~
_deps/zipper-src/zipper/unzipper.cpp:197:9: error: 'FILETIME' was not declared in this scope; did you mean 'ETIME'?
  197 |         FILETIME ftm, ftLocal, ftCreate, ftLastAcc, ftLastWrite;
      |         ^~~~~~~~
      |         ETIME
_deps/zipper-src/zipper/unzipper.cpp:199:9: error: 'hFile' was not declared in this scope; did you mean 'gzFile'?
  199 |         hFile = CreateFileA(filename.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
      |         ^~~~~
      |         gzFile
_deps/zipper-src/zipper/unzipper.cpp:199:47: error: 'GENERIC_READ' was not declared in this scope
  199 |         hFile = CreateFileA(filename.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
      |                                               ^~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:199:62: error: 'GENERIC_WRITE' was not declared in this scope
  199 |         hFile = CreateFileA(filename.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
      |                                                              ^~~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:199:86: error: 'OPEN_EXISTING' was not declared in this scope
  199 |         hFile = CreateFileA(filename.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
      |                                                                                      ^~~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:199:17: error: 'CreateFileA' was not declared in this scope
  199 |         hFile = CreateFileA(filename.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
      |                 ^~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:200:22: error: 'INVALID_HANDLE_VALUE' was not declared in this scope
  200 |         if (hFile != INVALID_HANDLE_VALUE)
      |                      ^~~~~~~~~~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:202:33: error: 'ftCreate' was not declared in this scope
  202 |             GetFileTime(hFile, &ftCreate, &ftLastAcc, &ftLastWrite);
      |                                 ^~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:202:44: error: 'ftLastAcc' was not declared in this scope
  202 |             GetFileTime(hFile, &ftCreate, &ftLastAcc, &ftLastWrite);
      |                                            ^~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:202:56: error: 'ftLastWrite' was not declared in this scope
  202 |             GetFileTime(hFile, &ftCreate, &ftLastAcc, &ftLastWrite);
      |                                                        ^~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:202:13: error: 'GetFileTime' was not declared in this scope
  202 |             GetFileTime(hFile, &ftCreate, &ftLastAcc, &ftLastWrite);
      |             ^~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:203:36: error: 'WORD' was not declared in this scope
  203 |             DosDateTimeToFileTime((WORD)(dosdate >> 16), (WORD)dosdate, &ftLocal);
      |                                    ^~~~
_deps/zipper-src/zipper/unzipper.cpp:203:64: error: expected ')' before 'dosdate'
  203 |             DosDateTimeToFileTime((WORD)(dosdate >> 16), (WORD)dosdate, &ftLocal);
      |                                  ~                             ^~~~~~~
      |                                                                )
_deps/zipper-src/zipper/unzipper.cpp:203:74: error: 'ftLocal' was not declared in this scope
  203 |             DosDateTimeToFileTime((WORD)(dosdate >> 16), (WORD)dosdate, &ftLocal);
      |                                                                          ^~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:203:13: error: 'DosDateTimeToFileTime' was not declared in this scope
  203 |             DosDateTimeToFileTime((WORD)(dosdate >> 16), (WORD)dosdate, &ftLocal);
      |             ^~~~~~~~~~~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:204:48: error: 'ftm' was not declared in this scope; did you mean 'tm'?
  204 |             LocalFileTimeToFileTime(&ftLocal, &ftm);
      |                                                ^~~
      |                                                tm
_deps/zipper-src/zipper/unzipper.cpp:204:13: error: 'LocalFileTimeToFileTime' was not declared in this scope
  204 |             LocalFileTimeToFileTime(&ftLocal, &ftm);
^~~~~~~~~~~~~~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:205:13: error: 'SetFileTime' was not declared in this scope
  205 |             SetFileTime(hFile, &ftm, &ftLastAcc, &ftm);
      |             ^~~~~~~~~~~
_deps/zipper-src/zipper/unzipper.cpp:206:13: error: 'CloseHandle' was not declared in this scope
  206 |             CloseHandle(hFile);
      |             ^~~~~~~~~~~
[160/233] Building CXX object _deps/cpr-build/cpr/CMakeFiles/cpr.dir/session.cpp.obj
[161/233] Building CXX object _deps/zipper-build/CMakeFiles/Zipper-static.dir/zipper/CDirEntry.cpp.obj
FAILED: _deps/zipper-build/CMakeFiles/Zipper-static.dir/zipper/CDirEntry.cpp.obj 
D:\a\_temp\msys\msys64\mingw64\bin\c++.exe  -DLINUX -DUSE_ZLIB -I_deps/zipper-src/minizip -I_deps/zipper-src/zipper -I_deps/zipper-src -I_deps/zipper-build -fno-strict-aliasing -fPIC -O3 -DNDEBUG   -std=gnu++11 -MD -MT _deps/zipper-build/CMakeFiles/Zipper-static.dir/zipper/CDirEntry.cpp.obj -MF _deps\zipper-build\CMakeFiles\Zipper-static.dir\zipper\CDirEntry.cpp.obj.d -o _deps/zipper-build/CMakeFiles/Zipper-static.dir/zipper/CDirEntry.cpp.obj -c _deps/zipper-src/zipper/CDirEntry.cpp
_deps/zipper-src/zipper/CDirEntry.cpp: In static member function 'static bool zipper::CDirEntry::createDir(const string&, const string&)':
_deps/zipper-src/zipper/CDirEntry.cpp:229:59: error: too many arguments to function 'int mkdir(const char*)'
  229 |     return (mkdir(Dir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) == 0);
      |                                                           ^
In file included from D:/a/_temp/msys/msys64/mingw64/x86_64-w64-mingw32/include/fcntl.h:8,
                 from _deps/zipper-src/zipper/defs.h:10,
                 from _deps/zipper-src/zipper/CDirEntry.cpp:17:
D:/a/_temp/msys/msys64/mingw64/x86_64-w64-mingw32/include/io.h:282:15: note: declared here
  282 |   int __cdecl mkdir (const char *) __MINGW_ATTRIB_DEPRECATED_MSVC2005;
      |               ^~~~~
```
I'm using msys2 under github actions, I'm following this [guide](https://www.msys2.org/docs/ci/)

--- a/zipper/CDirEntry.cpp
+++ b/zipper/CDirEntry.cpp
@@ -223,7 +223,7 @@ bool CDirEntry::createDir(const std::string & dir, const std::string & parent)
         createDir(actualParent);
     }
 
-#if defined(USE_WINDOWS)
+#if defined(USE_WINDOWS) || defined (__MINGW32__)
     return (mkdir(Dir.c_str()) == 0);
 #else
     return (mkdir(Dir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) == 0);

--- a/zipper/defs.h
+++ b/zipper/defs.h
@@ -10,7 +10,7 @@ extern "C"
 #include <fcntl.h>
 #include <sys/stat.h>
 
-#if (defined(__WIN32__) || defined(WIN32) || defined(_WIN32) || defined(_WIN64) || defined(_MSC_VER)) && !defined(CYGWIN)
+#if (defined(__WIN32__) || defined(WIN32) || defined(_WIN32) || defined(_WIN64) || defined(_MSC_VER)) && !defined(CYGWIN) && !defined(__MINGW32__)
 #  define USE_WINDOWS 1
 #endif
 
@@ -71,7 +71,7 @@ typedef struct stat STAT;
 #  endif
 #endif
 
-#if defined(USE_WINDOWS)
+#if defined(USE_WINDOWS) || defined (__MINGW32__)
 #  define MKDIR(d) _mkdir(d)
 #  define CHDIR(d) _chdir(d)
 #else

--- a/zipper/tools.cpp
+++ b/zipper/tools.cpp
@@ -8,7 +8,7 @@
 #include <cstdio>
 #include <iostream>
 
-#if (USE_WINDOWS)
+#if defined(USE_WINDOWS)
 #  include "tps/dirent.h"
 #  include "tps/dirent.c"
 #else

--- a/zipper/unzipper.cpp
+++ b/zipper/unzipper.cpp
@@ -192,7 +192,7 @@ public:
 
     void changeFileDate(const std::string& filename, uLong dosdate, tm_unz tmu_date)
     {
-#if (USE_WINDOWS)
+#if defined(USE_WINDOWS)
         HANDLE hFile;
         FILETIME ftm, ftLocal, ftCreate, ftLastAcc, ftLastWrite;
 

--- a/zipper/unzipper.cpp
+++ b/zipper/unzipper.cpp
@@ -192,7 +192,7 @@ public:
 
     void changeFileDate(const std::string& filename, uLong dosdate, tm_unz tmu_date)
     {
-#ifdef _WIN32
+#if (USE_WINDOWS)
         HANDLE hFile;
         FILETIME ftm, ftLocal, ftCreate, ftLastAcc, ftLastWrite;
 

